daemon/daemon.go

```diff
@@ -448,6 +448,11 @@ func (daemon *Daemon) registerLink(parent, child *container.Container, alias str
 
 // SetClusterProvider sets a component for querying the current cluster state.
 func (daemon *Daemon) SetClusterProvider(clusterProvider cluster.Provider) {
+	// Daemon is in charge of removing the attachable networks with
+	// connected containers when the node leaves the swarm
+	if clusterProvider == nil {
+		daemon.clearAttachableNetworks()
+	}
```

Comment:
True. But this is the only place in the _daemon_ package where we know the node is leaving the cluster.
And the cleanup has to be performed from the _daemon_.

The other option was to add a specific method to the `backend` interface to be called from the _cluster_ package, but I did not like it because it should be called in several places, while they all converge to eventually call `func (daemon *Daemon) SetClusterProvider(clusterProvider cluster.Provider) `, so to me this was the safest approach.